core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -239,6 +242,53 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
   }
 
+  val unresolvedStarRegex = """(cannot resolve ')(\w+).(\w+).*(' give input columns.*)""".r
+  val unresolvedColRegex = """(cannot resolve '`)(\w+).(\w+).(\w+)(.*given input columns.*)""".r
+
+  // Hack to fix SNAP-2440 ( TODO: Will return after 102 to see if a better fix is warranted )
+  private def reAnalyzeForUnresolvedAttribute(
+    originalPlan: LogicalPlan, e: AnalysisException,
+    retryCount: Int): Option[LogicalPlan] = {
+    if (!e.getMessage().contains("cannot resolve")) return None
+    val unresolvedNodes = originalPlan.expressions.filter(
+      x => x.isInstanceOf[UnresolvedStar] | x.isInstanceOf[UnresolvedAttribute])
+    if (retryCount > unresolvedNodes.size) return None
```

Comment:
This can be merged with the first statement of the method to return.

Reply:
See, if the exception type is not "cannot resolve" in the message you need not traverse the tree and that is why I kept it like that.

Reply:
ok.